### PR TITLE
Fixed classpath bug

### PR DESCRIPTION
**Pull request by gzoller**
@smarter just posted a PR that fixed a classpath issue in QuoteDriver.  The exact same code is also in ConsumeTasty (compiler subdir), and it's been causing problems in Tasty inspection.  I cut/paste @smarter's fix into ConsumeTasty so it will work the same way.   I didn't refactor to avoid DRY because this is my first commit and I don't feel confident I understand the project structure yet.

I've signed the CLA.

### CODE REVIEW

File: compiler/src/dotty/tools/dotc/consumetasty/ConsumeTasty.scala [hunk 1]
```diff
@@ -18,23 +23,33 @@ object ConsumeTasty {
...
+    @tailrec
+    def loop(cl: ClassLoader, suffixClasspath: String): String =
+      cl match {
+        case cl: URLClassLoader =>
+          val updatedClasspath = cl.getURLs
+            .map(url => Paths.get(url.toURI).toAbsolutePath.toString)
+            .mkString(
+              "",
+              File.pathSeparator,
+              if (suffixClasspath.isEmpty) "" else File.pathSeparator + suffixClasspath
```
Comment (nicolasstucki) [inline compiler/src/dotty/tools/dotc/consumetasty/ConsumeTasty.scala]: The current version in #7953 has been updated. You should take the current version.

Comment (gzoller) [reply compiler/src/dotty/tools/dotc/consumetasty/ConsumeTasty.scala]: Hi, Nicolas.  I'm not sure what you mean.  I looked at #7953, which is now merged into master.  ConsumeTasty.scala in master still looks like the old way of getting the class path, not @smarter's new way.  What change is requested?  Thx.

Comment (nicolasstucki) [reply compiler/src/dotty/tools/dotc/consumetasty/ConsumeTasty.scala]: I meant, to copy the code in https://github.com/lampepfl/dotty/pull/7953/files#diff-0cf1d278cf2ee8dca0cb2b7bcb62348aR75

Comment (odersky) [reply compiler/src/dotty/tools/dotc/consumetasty/ConsumeTasty.scala]: Should we take this as a sign to refactor into one method?

Comment (nicolasstucki) [reply compiler/src/dotty/tools/dotc/consumetasty/ConsumeTasty.scala]: Yes, but I would leave it for a future PR. I have a bigger refactoring coming that will address that.

Comment (nicolasstucki) [reply compiler/src/dotty/tools/dotc/consumetasty/ConsumeTasty.scala]: I opened #8020 for this refactoring.

**Comment by nicolasstucki** [comment]
@gzoller I decided to merge it as it does fix the current issue.

**Comment by gzoller** [comment]
Ok, thanks!  Will delete my fork.

**Pull request merged**